pkg/util/hostname/kubelet/hostname.go

```diff
@@ -36,10 +36,10 @@ func HostnameProvider(ctx context.Context, options map[string]interface{}) (stri
 		return "", fmt.Errorf("couldn't fetch the host nodename from the kubelet: %s", err)
 	}
 
-	clusterName := clustername.GetClusterName(ctx, nodeName)
+	clusterName := clustername.GetRFC1123CompliantClusterName(ctx, nodeName)
```

Comment:
We only need a `GetRFC1123CompliantClusterName` if we use it as a hostAlias, else we can keep it as it is (tags). This makes sure that `hosts` can be linked with the `hostAlias`